src/operator/rnn-inl.h

```diff
@@ -361,9 +361,9 @@ void RNNBackward(DType* ws,
                  DType* rs,
                  const int num_layers,
                  const int direction,
-                 const int seq_length,
-                 const int batch_size,
-                 const int input_size,
+                 const index_t seq_length,
+                 const index_t batch_size,
+                 const index_t input_size,
```

Comment:
size_t? if its not a breaking chnage

Reply:
I think we can keep the signed index_t since all the functions being called using signed and the omp loop requires a signed index as well.

Reply:
Agree with keeping the signed `index_t` :100:

Reply:
I agree @apeforest! I believe the omp loop’s required signed index was the root cause of the segfault when I made the size_t changes.